### PR TITLE
Feature/Update-table-for-getting-PBFW-clients

### PR DESCRIPTION
**Pull request by nobert-mumo**
This PR updates various facts based on logic from the new pbfw intermediate table.

### CODE REVIEW

File: Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql [hunk 1]
```diff
@@ -9,20 +9,24 @@ BEGIN
...
-                    Sitecode,
-                    Visitdate
-             FROM   Ods.Dbo.Mnch_ancvisits
-             WHERE  Hivstatusbeforeanc = 'KP'
-                     OR Hivtestfinalresult = 'Positive'
-		),
+              select Row_number()
+                            OVER (
+                            Partition BY Patientpk, Sitecode
+                            ORDER BY Visitdate asc ) as num,
```
Comment (Marymary-dev) [inline Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:15]: @nobert-mumo  Are we still using data from MNCH in the new pbfw models?

**Review by Marymary-dev — COMMENTED** [review]
@nobert-mumo  Are we still using data from MNCH in the new pbfw models?

File: Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql (hunk 1 above)
Comment (nobert-mumo) [reply Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:15]: @Marymary-dev yeah indeed we still use some info from MNCH,
The actual logic change for this fact table is the intermediate table we use for selecting PBFWs.

But for ANC1 we coalesce the anc inferred from greencard & the one from MNCH.
See here: https://github.com/palladiumkenya/dwh-etl/blob/4d07df649b0f5600fad6466702c4a5d53147a397/Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql#L205

But I was wondering if we need to do the same for ANC2, ANC3, ANC4 ?

File: Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql [hunk 2]
```diff
@@ -151,13 +155,54 @@ BEGIN
...
+              SELECT
+                    Patient.PatientPKHash,
                     Patient.Sitecode,
-                    Dob,
-                    Gender,
-                    coalesce(GreenCardAncDate1, Ancdate1.ANCDate1) AS ANCDate1,
+                    dim_patient.Dob,
+                    dim_patient.Gender,
+                    coalesce(earliest_anc_from_greencard.GreenCardAncDate1, Ancdate1.ANCDate1) AS ANCDate1,
                     ANCDate2.Ancdate2,
```
Comment (Marymary-dev) [inline Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:206]: @nobert-mumo  yes have left the earliest_anc from anc where it is the cecond one if visits are in asc order

Comment (nobert-mumo) [reply Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:206]: @Marymary-dev you mean we do for the other ANCs dates as we have done for ANC1 ?

Comment (Marymary-dev) [reply Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:206]: @nobert-mumo  yes, but before which variables except KP and NP are we deriving from the ANC data ?

File: Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql [hunk 3]
```diff
@@ -203,42 +264,71 @@ BEGIN
...
+              left join ODS.dbo.CT_ARTPatients as art 
+                                   on art.PatientPK = Patient.PatientPK
+                            and art.SiteCode = Patient.SiteCode
+              left join latest_anc        
+                            on latest_anc.PatientPK = Patient.PatientPK 
+                            and latest_anc.SiteCode = Patient.SiteCode
+              left join earliest_anc_from_greencard 
+                            on earliest_anc_from_greencard.PatientPK = Patient.PatientPK 
+                            and earliest_anc_from_greencard.SiteCode = Patient.SiteCode
+              left join second_anc_greencard
```
Comment (nobert-mumo) [inline Scripts/NDWH/PMTCT_FACT_TABLES/FactPBFW.sql:322]: @Marymary-dev I have now added the extra joins for ANC visits from green card.

**Review by DennisGibz — APPROVED** [review]

**Review by Marymary-dev — APPROVED** [review]

**Pull request merged**